osu.Game/Screens/Edit/Compose/Components/DragBox.cs

```diff
@@ -111,5 +102,75 @@ public override void Hide()
         public override void Show() => State = Visibility.Visible;
 
         public event Action<Visibility> StateChanged;
+
+        public class BoxWithBorders : CompositeDrawable
+        {
+            private readonly LayoutValue cache = new LayoutValue(Invalidation.RequiredParentSizeToFit);
+
+            public BoxWithBorders()
+            {
+                AddLayout(cache);
+            }
+
+            protected override void Update()
+            {
+                base.Update();
+
+                if (!cache.IsValid)
+                {
+                    createContent();
+                    cache.Validate();
+                }
+            }
+
+            private void createContent()
+            {
+                if (DrawSize == Vector2.Zero)
+                {
+                    ClearInternal();
+                    return;
+                }
+
+                // Make lines the same width independent of display resolution.
+                float lineThickness = DrawWidth > 0
+                    ? DrawWidth / ScreenSpaceDrawQuad.Width * 2
+                    : DrawHeight / ScreenSpaceDrawQuad.Height * 2;
+
+                Padding = new MarginPadding(-lineThickness);
```

Comment:
You've mentioned on stream that you would expect the borders to "fold onto" each other when a selection has zero width or height. This can be achieved by setting the padding to half of line thickness:

```suggestion
                Padding = new MarginPadding(-lineThickness / 2);
```

Before:

https://user-images.githubusercontent.com/20418176/167127603-c9a3ae0b-726f-4c3f-a374-1374371ae270.mp4

After:

https://user-images.githubusercontent.com/20418176/167127528-1e6186d0-02da-4670-88bd-4c6568b58fab.mp4

Reply:
looks better, feel free to apply the patch